### DISCUSSION TITLE
Mutual exclusive subcommand and ArgGroup

### DISCUSSION THREAD

**Discussion by Onedy** [remkop/picocli#1740] [Q&A] [answered]
Hi,

Is there a way to create a mutual exclusion between a subcommand and an ArgGroup?

I'd like to have these two commands:
`stats stop`
`stats <tweetUrl> [-l]`

Currently I have tried this class:

```
@CommandLine.Command(name = "stats", description = "Stats related operations")
public class StatsCommand implements Runnable {

    @CommandLine.ArgGroup(exclusive = true, multiplicity = "1")
    private ExclusiveArgs what;

    static class ExclusiveArgs {

        @CommandLine.Command(name = "stop", description = "Stops current live stats")
        void stopLive() { }

        @CommandLine.ArgGroup(exclusive = false)
        ExclusiveArgs.DependentArgs dependentArgs;

        static class DependentArgs {

            @Parameters(arity = "1", description = "Tweet URL to get stats of")
            private String tweetUrl;

            @CommandLine.Option(names = {"-l", "--live"})
            private boolean live;
        }
    }

    @Override
    public void run() { }
}
```
I also tried replacing the stopLive() method with a static inner class which implemented Runnable, same result.

**Comment by remkop** [comment, marked as answer]
> Is there a way to create a mutual exclusion between a subcommand and an ArgGroup?

No, not really. But then: do you really need to? (And for that matter, do you really need to use an ArgGroup?)

If you have a parent command with options/parameters and a subcommand, then the user may specify any combination.

```
stats stop
stats <tweetUrl> [-l]
stats <tweetUrl> [-l] stop
```

Is this really a problem?
The application could just choose to ignore input that is not applicable...

And, if it is a problem, then you can add validation logic in the `stopLive()` method and throw a `ParameterException` like described in the [custom validation](https://picocli.info/#_custom_validation) section of the manual.
